pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -478,14 +577,17 @@ func (o *EditOptions) visitToCreate(createVisitor resource.Visitor) error {
 func (o *EditOptions) visitAnnotation(annotationVisitor resource.Visitor) error {
 	// iterate through all items to apply annotations
 	err := annotationVisitor.Visit(func(info *resource.Info, incomingErr error) error {
-		// put configuration annotation in "updates"
-		if o.ApplyAnnotation {
-			if err := kubectl.CreateOrUpdateAnnotation(true, info, o.Encoder); err != nil {
+		if o.Record || cmdutil.ContainsChangeCause(info) {
```

Comment:
why the reordering? the apply annotation is more important than the change cause, which is why we recorded it first.